sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/options/ShareFileSeekableByteChannelWriteOptions.java

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.storage.file.share.options;
+
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.storage.file.share.models.FileLastWrittenMode;
+import com.azure.storage.file.share.models.ShareRequestConditions;
+
+import java.util.Objects;
+
+/**
+ * Options for obtaining a {@link java.nio.channels.SeekableByteChannel} backed by an Azure Storage Share File.
+ */
+public final class ShareFileSeekableByteChannelWriteOptions {
+    private static final ClientLogger LOGGER = new ClientLogger(ShareFileSeekableByteChannelWriteOptions.class);
+
+    /**
+     * Mode to open the channel for writing.
+     */
+    public enum WriteMode {
+        /**
+         * Opens channel to an existing file for writing.
+         */
+        WRITE,
+
+        /**
+         * Creates a new file for writing and opens the channel. If the file already exists, it will be overwritten.
+         * Requires a value be set with {@link ShareFileSeekableByteChannelWriteOptions#setFileSize(Long)}.
+         */
+        OVERWRITE
+    }
+
+    private final WriteMode channelMode;
+    private Long fileSize;
+    private ShareRequestConditions requestConditions;
+    private FileLastWrittenMode fileLastWrittenMode;
+
+    /**
+     * Options constructor.
+     * @param mode What usage mode to open the channel in.
+     */
+    public ShareFileSeekableByteChannelWriteOptions(WriteMode mode) {
+        channelMode = Objects.requireNonNull(mode, "'mode' cannot be null.");
+    }
+
+    /**
+     * This parameter is required when this instance is configured to {@link WriteMode#OVERWRITE}.
+     * @return Usage mode to be used by the resulting channel.
+     */
+    public WriteMode getChannelMode() {
+        return channelMode;
+    }
+
+
+    /**
+     * This parameter is required when this instance is configured to {@link WriteMode#OVERWRITE}.
+     * @return New size of the target file.
+     */
+    public Long getFileSize() {
+        return fileSize;
+    }
+
+    /**
+     * @param fileSize New size of the target file.
+     * @return The updated instance.
+     * @throws UnsupportedOperationException When setting a file size on options that don't create a new file.
+     */
+    public ShareFileSeekableByteChannelWriteOptions setFileSize(Long fileSize) {
+        if (channelMode != WriteMode.OVERWRITE) {
+            throw LOGGER.logExceptionAsError(
+                new UnsupportedOperationException("Cannot set 'fileSize' unless creating a new file."));
+        }
+        this.fileSize = fileSize;
```

Comment:
Should this throw an exception if file size is less than zero? Or less than or equal to zero?